n_helpers.c

```diff
@@ -435,39 +514,40 @@ const char * NoteBinaryTransmit(uint8_t * data, size_t dataLen, size_t bufLen,
     // Validate the index provided by the caller, against the `length` value
     // returned from the Notecard to ensure the caller and Notecard agree on
     // how much data is residing on the Notecard.
-    if (offset != len) {
+    if ((long)notecardOffset != len) {
```

Comment:
Is is theoretically possible for this to get us into trouble? Thinking about if notecardOffset has the MSB set and long is 32-bits.

Reply:
`len` comes from `JGetInt` which returns a long. So we would be in trouble either way.

If the underlying hardware were able to support > 2MB of flash, then we would immediately have a problem with CJSON, so the first signs of fire would not be at the level of this API.